tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -1059,11 +1059,35 @@ static void test_nanocoap__token_length_over_limit(void)
     TEST_ASSERT_EQUAL_INT(8, coap_get_token_len(&pkt));
     TEST_ASSERT_EQUAL_INT(0, pkt.payload_len);
 
-    /* Invalid packet (TKL = 9) */
+    /* Invalid packet (TKL = 15) */
     res = coap_parse(&pkt, buf_invalid, sizeof(buf_invalid));
     TEST_ASSERT_EQUAL_INT(-EBADMSG, res);
 }
 
+/*
+ * Verifies that coap_parse() recognizes token length bigger than allowed.
+ */
+static void test_nanocoap__token_length_ext(void)
+{
+    const char *token = "0123456789ABCDEF";
+
+    uint8_t buf[32];
+    coap_hdr_t *hdr = (void *)buf;
+
+    TEST_ASSERT_EQUAL_INT(21, coap_build_hdr(hdr, COAP_TYPE_CON,
+                                             (void *)token, strlen(token),
+                                             COAP_CODE_204, 23));
+    coap_pkt_t pkt;
+    int res = coap_parse(&pkt, buf, 21);
+
+    TEST_ASSERT_EQUAL_INT(0, res);
+    TEST_ASSERT_EQUAL_INT(204, coap_get_code(&pkt));
+    TEST_ASSERT_EQUAL_INT(23, coap_get_id(&pkt));
+    TEST_ASSERT_EQUAL_INT(strlen(token), coap_get_token_len(&pkt));
+    TEST_ASSERT_EQUAL_INT(0, memcmp(coap_get_token(&pkt), token, strlen(token)));
+    TEST_ASSERT_EQUAL_INT(0, pkt.payload_len);
```

Comment:
Maybe cover the TKL directly? `TEST_ASSERT_EQUAL_INT(13, pkt.hdr.ver_t_tkl & 0xf);` (not sure if this would work :angel:)